extensions/vertx/runtime/src/main/java/io/quarkus/vertx/core/runtime/QuarkusExecutorFactory.java

```diff
@@ -28,20 +30,51 @@ public QuarkusExecutorFactory(VertxConfiguration conf, LaunchMode launchMode) {
 
     @Override
     public ExecutorService createExecutor(ThreadFactory threadFactory, Integer concurrency, Integer maxConcurrency) {
+        // The current Vertx impl creates two external executors during initialization
+        // The first one is used for the worker thread pool, the second one is used internally,
+        // and additional executors may be created on demand
+        // Unfortunately, there is no way to distinguish the particular executor types
+        // Therefore, we only consider the first one as the worker thread pool
```

Comment:
This is IMO quite risky but the Vertx API does not provide a way to distinguish the particular executor types... :shrug:

Reply:
The internal one is for file system operations only.

Reply:
ok, I think it was already broken... 

We could change the Vert.x SPI to get the "type" (worker, internal, custom)

Reply:
> We could change the Vert.x SPI to get the "type" (worker, internal, custom)

Yes, that would be helpful.

CC @jponge